artiq/protocols/pc_rpc.py

```diff
@@ -487,6 +487,17 @@ def __init__(self, targets, description=None, builtin_terminate=False,
         else:
             self._noparallel = asyncio.Lock()
 
+    @staticmethod
+    def _document_function(function: typing.Callable) -> typing.Tuple[dict, str]:
```

Comment:
Please remove the type annotations.

Reply:
Can you explain why the type annotations should be removed? Other than not being used elsewhere in ARTIQ (https://github.com/m-labs/artiq/pull/1181#discussion_r227604791)?

There is no downside to using type annotations, it makes the code more understandable, and it helps IDEs with autocomplete. Guido van Rossum supports them, and they are making their way into more projects as Python2 support is dropped. Some projects that use them:

* Pylint: https://github.com/PyCQA/pylint/issues/2079
* Black (formatter): https://github.com/ambv/black/blob/master/black.py

Overview of type hints and their benefits; https://stackoverflow.com/questions/32557920/what-are-type-hints-in-python-3-5

Reply:
They are not used anywhere else in ARTIQ, they are not used by most other Python projects, they are difficult to write, some functions may return different types depending on the context, and annotations are ignored by the interpreter and are (currently at least) not checked automatically.

Reply:
Discussion continued below in https://github.com/m-labs/artiq/pull/1188#issuecomment-451571026